cmd/test/test.go

```diff
@@ -165,60 +165,74 @@ func newTestCommand() *cobra.Command {
+						outputFilePath = parts[1]
 					}
+					if format == OutputAppstudio {
+						// The appstudio format is unknown to Conftest so we handle it ourselves
 
-					if outputFilePath != "" {
-						err := os.WriteFile(outputFilePath, reportOutput, 0600)
+						if resultsErr != nil {
+							return appstudioErrorHandler(runner.NoFail, "running test", resultsErr)
+						}
+
+						report := appstudioReport(results, runner.Namespace)
+						reportOutput, err := json.Marshal(report)
 						if err != nil {
-							return fmt.Errorf("creating output file: %w", err)
+							return appstudioErrorHandler(runner.NoFail, "output results", err)
 						}
-					}
-					fmt.Fprintln(cmd.OutOrStdout(), string(reportOutput)+"\n")
 
-				} else {
-					// Conftest handles the output
+						if outputFilePath != "" {
+							err := os.WriteFile(outputFilePath, reportOutput, 0600)
+							if err != nil {
+								return fmt.Errorf("creating output file: %w", err)
+							}
+						}
+						fmt.Fprintln(cmd.OutOrStdout(), string(reportOutput)+"\n")
 
-					if resultsErr != nil {
-						return fmt.Errorf("running test: %w", resultsErr)
-					}
+					} else {
+						// Conftest handles the output
 
-					var outputFile *os.File
-					if outputFilePath != "" {
-						outputFile, err = os.Create(outputFilePath)
-						if err != nil {
-							return fmt.Errorf("creating output file: %w", err)
+						if resultsErr != nil {
+							return fmt.Errorf("running test: %w", resultsErr)
 						}
-						defer outputFile.Close()
-					}
 
-					outputter := output.Get(runner.Output, output.Options{
-						NoColor:            runner.NoColor,
-						SuppressExceptions: runner.SuppressExceptions,
-						Tracing:            runner.Trace,
-						JUnitHideMessage:   viper.GetBool("junit-hide-message"),
-						File:               outputFile,
-					})
-					if err := outputter.Output(results); err != nil {
-						return fmt.Errorf("output results: %w", err)
-					}
+						var outputFile *os.File
+						if outputFilePath != "" {
+							outputFile, err = os.Create(outputFilePath)
+							if err != nil {
+								return fmt.Errorf("creating output file: %w", err)
+							}
+							defer outputFile.Close()
+						}
 
-					if outputFilePath != "" {
-						contents, err := os.ReadFile(outputFile.Name())
-						if err != nil {
-							return fmt.Errorf("copying output file to stdout: %w", err)
+						outputter := output.Get(format, output.Options{
+							NoColor:            runner.NoColor,
+							SuppressExceptions: runner.SuppressExceptions,
+							Tracing:            runner.Trace,
+							JUnitHideMessage:   viper.GetBool("junit-hide-message"),
+							File:               outputFile,
+						})
+						if err := outputter.Output(results); err != nil {
+							return fmt.Errorf("output results: %w", err)
+						}
+
+						if outputFilePath != "" {
+							contents, err := os.ReadFile(outputFile.Name())
+							if err != nil {
+								return fmt.Errorf("copying output file to stdout: %w", err)
```

Comment:
Hold on, I think there's a requirements misunderstanding. If the user specifies a file, then we should write the output to that file only. It looks like you're doing extra work to also print it on stdout.

Reply:
@simonbaird Yeah, that's a leftover from the old implementation. I'll adjust.

Reply:
I think you can just delete line 229 to 235 or so.